### ISSUE TITLE
simple_service_discovery.ipynb notebook gives warnings in cell#2 (related to Issue#257 ?)

### ISSUE THREAD

**Issue by KenMighell** [astropy/pyvo#270]
```pyvo.__version__='1.1'``` 

The Jupyter notebook  ```pyvo/examples/notebooks/simple_service_discovery.ipynb```  issues warnings in cell#2. 

This problem _may_ be related to Issue #257 .

This script

```
#!/usr/bin/env python3
import pyvo
services = pyvo.regsearch(servicetype="sia", waveband="x-ray")
# EOF
```
produces these warnings

```
WARNING: UnknownElementWarning: None:3:1106: UnknownElementWarning: Unknown element mirrorURL [pyvo.utils.xml.elements]
WARNING: UnknownElementWarning: None:125:4415: UnknownElementWarning: Unknown element mirrorURL [pyvo.utils.xml.elements]
WARNING: UnknownElementWarning: None:125:4732: UnknownElementWarning: Unknown element mirrorURL [pyvo.utils.xml.elements]
WARNING: UnknownElementWarning: None:125:5044: UnknownElementWarning: Unknown element mirrorURL [pyvo.utils.xml.elements]
WARNING: UnknownElementWarning: None:125:5344: UnknownElementWarning: Unknown element mirrorURL [pyvo.utils.xml.elements]
```

Inserting these two lines after importing pyvo stops the warnings from being displayed

```
import warnings
warnings.filterwarnings("ignore", module="pyvo.utils.xml.*")
```

but at the cost of losing all XML parsing diagnostics.

**Comment by KenMighell** [comment]
===== More notebook problems =====

(1) The other Jupyter notebook ```pyvo/examples/notebooks/working_with_registry_results.ipynb``` gives similar warnings (```Unknown element mirrorURL```) in three different cells: cell#1, cell#5, and cell#7.

(2) The code in cell#6:
```
nvss = vo.registry.ivoid2service('ivo://nasa.heasarc/skyview/nvss',servicetype='sia')
nvss.search(pos=(350.85, 58.815),size=0.25,format="image/fits").to_table()
```
gives the following error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-892a8a3f83d6> in <module>
----> 1 nvss = vo.registry.ivoid2service('ivo://nasa.heasarc/skyview/nvss',servicetype='sia')
      2 nvss.search(pos=(350.85, 58.815),size=0.25,format="image/fits").to_table()

/opt/anaconda3/lib/python3.8/site-packages/pyvo/registry/regtap.py in ivoid2service(ivoid, servicetype)
    445     }
    446     for result in results:
--> 447         thistype = result["standard_id"].decode()
    448         if thistype not in ivo_cls.keys():
    449             # This one is not a VO service

AttributeError: 'str' object has no attribute 'decode'
```
Note that this notebook was last updated 2 years ago.

**Comment by andamian** [comment]
@KenMighell thanks for bringing this to our attention. I believe this is related to #268 for which there's already a PR (#269) that gets rid of these warnings.

(2) in the notebook needs to be investigated and addressed - we'll keep this issue open to track that.

**Comment by KenMighell** [comment]
In the Jupyter notebook ```pyvo/examples/notebooks/simple_service_discovery.ipynb```, warnings previously found in cell#2 no longer appear when using version 1.2.dev701 (branch: 6817404).

The definition of ```__all__``` in pyvo/io/vosi/voresource.py should probably be appended to include the new MirrorURL class:

`__all__ = [
    "ValidationLevel", "Capability", "Interface", "AccessURL", 
    "MirrorURL",
    "SecurityMethod", "WebBrowser", "WebService"]
`

**Comment by msdemlei** [comment]
On Mon, Aug 30, 2021 at 12:49:42PM -0700, KenMighell wrote:
> In the Jupyter notebook ```pyvo/examples/notebooks/simple_service_discovery.ipynb```, warnings previously found in cell#2 no longer appear when using version 1.2.dev701 (branch: 6817404).
> 
> The definition of ```__all__``` in pyvo/io/vosi/voresource.py should probably be appended to include the new MirrorURL class:
> 
> `__all__ = [
>     "ValidationLevel", "Capability", "Interface", "AccessURL", 
>     "MirrorURL",
>     "SecurityMethod", "WebBrowser", "WebService"]
> `

Fixed PR #269 accordingly.

**Comment by KenMighell** [comment]
Solution to the bug that crashes cell#6 of the PyVO Jupyter notebook  
```pyvo/examples/notebooks/working_with_registry_results.ipynb``` :

The bottom of the function ```ivoid2service()``` in the file ```pyvo/pyvo/registry/regtap.py``` calls a ```decode()``` method 3 times:  

thistype = result["standard_id"].decode()   

return(cls(result["access_url"].decode()))   

services.append(cls(result["access_url"].decode())) 

The objects to be "decoded" are type str which has no decode() method in Python3;  str types had a decode() method in Python2.

***BUG FIX*** Deleting the code ```.decode()``` all 3 times that it appears in the function ```ivoid2service()``` fixes the bug.

**Comment by pllim** [comment]
I think this was fixed in #276.
